src/format.rs

```diff
@@ -210,6 +210,44 @@ impl<'a, W: Write> Collector for FormatCollector<'a, W> {
         Ok(())
     }
 
+    #[inline]
+    fn nanosecond_of_minute(&mut self) -> Result<(), Self::Error> {
+        let nanoseconds = self.time.nanosecond();
+
+        let keep_digits: usize = if nanoseconds % 10 != 0 {
+            9
+        } else if (nanoseconds / 10) % 10 != 0 {
+            8
+        } else if (nanoseconds / 100) % 10 != 0 {
+            7
+        } else if (nanoseconds / 1_000) % 10 != 0 {
+            6
+        } else if (nanoseconds / 10_000) % 10 != 0 {
+            5
+        } else if (nanoseconds / 100_000) % 10 != 0 {
+            4
+        } else if (nanoseconds / 1_000_000) % 10 != 0 {
+            3
+        } else if (nanoseconds / 10_000_000) % 10 != 0 {
+            2
+        } else if (nanoseconds / 100_000_000) % 10 != 0 {
+            1
+        } else {
+            0
```

Comment:
Should this be `1` so that if nanosec part was `0` we output `0`?
Or actually, I feel like we should just output 9 digits anyway like what `%N` of ruby does.

Reply:
Ok, adjusted to output 9 digits and added a test case

Reply:
Sorry, I meant to output 9 digits regardless of the input nanoseconds precision, e.g. `.125000000`.
This is what ruby does for `%N`
```
[10] pry(main)> Time.at(0.125).strftime('%N')
=> "125000000"
```
and here's Python does for `%f`
```
>>> datetime.datetime.fromtimestamp(0.125).strftime('%f')
'125000'
```

Reply:
No worries, that should simplify the implementation a lot 👍